src/renderer/component/publishForm/internal/license-type.jsx

```diff
@@ -0,0 +1,145 @@
+// @flow
+import * as React from "react";
+import { FormRow, FormField } from 'component/common/form';
+import Button from 'component/link';
+
+type Props = {
+  licenseType: string,
+  copyrightNotice: ?string,
+  licenseUrl: ?string,
+  otherLicenseDescription: ?string,
+  handleLicenseChange: (string, string) => void,
+  handleLicenseDescriptionChange: (SyntheticInputEvent<*>) => void,
+  handleLicenseUrlChange: (SyntheticInputEvent<*>) => void,
+  handleCopyrightNoticeChange: (SyntheticInputEvent<*>) => void,
+}
+
+class LiscenseType extends React.PureComponent<Props> {
+  constructor() {
+    super();
+
+    (this: any).handleLicenseOnChange = this.handleLicenseOnChange.bind(this);
+  }
+
+  handleLicenseOnChange(event: SyntheticInputEvent<*>) {
+    const { handleLicenseChange } = this.props;
+    // $FlowFixMe
+    const { options, selectedIndex } = event.target;
+
+    const selectedOption = options[selectedIndex];
+    const licenseType = selectedOption.value;
+    const licenseUrl = selectedOption.getAttribute("data-url");
+
+    handleLicenseChange(licenseType, licenseUrl);
+  }
+
+  render() {
+    const {
+      licenseType,
+      otherLicenseDescription,
+      licenseUrl,
+      copyrightNotice,
+      handleLicenseChange,
+      handleLicenseDescriptionChange,
+      handleLicenseUrlChange,
+      handleCopyrightNoticeChange,
+     } = this.props;
+
+    const ccBy = __('Creative Commons Attribution 4.0 International');
+    const ccBySa = __('Creative Commons Attribution-ShareAlike 4.0 International');
+    const ccByNd = __('Creative Commons Attribution-NoDerivatives 4.0 International');
+    const ccByNc = __('Creative Commons Attribution-NonCommercial 4.0 International');
+    const ccByNcSa = __('Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International');
+    const ccByNcNd = __('Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International');
+
+    return (
+      <div className="card__content">
+        <FormField
+          label={__("License (Optional)")}
+          type="select"
+          value={licenseType}
+          onChange={this.handleLicenseOnChange}
+        >
+          <option>{__('None')}</option>
+          <option value="Public Domain">{__('Public Domain')}</option>
+          <option
```

Comment:
This could probably be done by iterating over a data structure rather than enumerating all of them